### PR TITLE
CI: fix npdev build

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] xref #41632
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

hopefully numpy will provide an alternative that makes this unnecessary

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@jbrockmendel you are changing the code here for unreleased compat issues. IIUC numpy may make some further changes. would it be better to add something to the 54 failing tests for now?

**Comment by jbrockmendel** [comment]
I'm holding out hope that numpy will handle this upstream, xref https://github.com/numpy/numpy/issues/19078

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -1583,7 +1584,9 @@ def searchsorted(arr, value, side="left", sorter=None) -> np.ndarray:
         # and `value` is a pd.Timestamp, we may need to convert value
         arr = ensure_wrapped_if_datetimelike(arr)
 
-    return arr.searchsorted(value, side=side, sorter=sorter)
+    with catch_warnings():
+        filterwarnings("ignore", category=FutureWarning)
```
Comment (jreback) [inline pandas/core/algorithms.py:1588]: why don't u use create_ndarray here?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/algorithms.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/algorithms.py:1588]: bc `value` can be a scalar here

**Comment by jbrockmendel** [comment]
@seberg should we just suppress these warnings for the time being in expectation of a fix/change upstream?

**Comment by seberg** [comment]
Sorry about being away for the long weekend.  I will look into this again, but I got the feeling that you will likely need some type of helper like `create_ndarray` in any case.  Hopefully, we can avoid the warning context at least, though.

**Comment by seberg** [comment]
We just discussed this in the numpy call, and I think the most likely solution will be to (unfortunately) delay the deprecation, to give you more time and hopefully a better solution to this.

That would mean you don't have to worry about string promotion changes for this release.

**Comment by jbrockmendel** [comment]
OK.  Is there anything I/we can do short(ish)-term to make the process easier on your end?

**Comment by seberg** [comment]
@jorisvandenbossche not sure.  Maybe we can meet up briefly and discuss how well `np.discover_array_params` would help you (I am very flexible)?  And also what pandas needs right now in terms of dtype/descriptor discovery from NumPy?
(Plus, I guess what pandas actually does itself, since you currently need those multiple passes before and after the NumPy pass.)

I have some hopes that we could both clean up pandas and make NumPy and pandas work better together in one go with just small modifications to NumPy (and exposure of the new API).  And that would just solve this issue incidentally.

**Comment by jbrockmendel** [comment]
> Maybe we can meet up briefly and discuss

Sure.  I'm in UTC-7, can do any AM between now and Tuesday, inclusive.

**Comment by seberg** [comment]
Great! Send you an invite, in case anyone is curious, put the details here also: https://hackmd.io/fe_VHCBjSwutSwhpPP8KSg

**Pull request closed without merging**